src/track/globaltrackcache.cpp

```diff
@@ -486,7 +506,7 @@ TrackPointer GlobalTrackCache::revive(
                 << plainPtr;
     }
     DEBUG_ASSERT(weakPtrRef.expired());
-    strongPtr = TrackPointer(plainPtr, evictAndSaveCachedTrack);
+    strongPtr = TrackPointer(plainPtr, EvictAndSaveFunctor(i->second.second));
```

Comment:
Ok, I understand. This is why we need to store the deleting pointer. Otherwise we would end up with different instances and disconnected reference counts for the same plain pointer. Maybe you could leave note here that it is crucial to use only one deleting pointer behind the scenes.